File: problems/e/everywhere/everywhere.py
#! /usr/bin/python3
"""
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
"""

def main():
    t = int(input())
    for _ in range(t):
        n = int(input())

        set_of_cities = set()
        for _ in range(n):
            set_of_cities.add(input())
        print(len(set_of_cities))


if __name__ == '__main__':
    main()

File: problems/e/everywhere/everywhere.cpp
/*
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
*/

#include <iostream>

using namespace std;

int main() {
    int t, n, total_cities;
    bool repeated_city;
    
    cin >> t;

    for (int i=0; i<t; i++) {
        cin >> n;
        string cities[n];
        for (int j=0; j<n; j++) {
            cin >> cities[j];
        }
        total_cities = 1;
        for (int j=1; j<n; j++) {
            repeated_city = false;
            for (int k=0; k<j; k++) {
                if (cities[j] == cities[k]) {
                    repeated_city = true;
                    break;
                }
            }
            if (repeated_city == false) {
                total_cities++;
            }
        }
        cout << total_cities << endl;
    }
}